### DISCUSSION TITLE
Not extruding during print

### DISCUSSION THREAD

**Discussion by vncntv** [moonglow/FlashForge_Marlin#47] [Q&A] [answered]
I have a flashforge dreamer with dual extruders and installed the prebuilt color UI (the one without any changes like swap etc...)
Everything works fine, and i can load and unload filament, the printhead extrudes filament just fine in that case (so i presume it is not jammed).

But while printing i see the extruder only produces very small sporadic blobs.
The start gcode has a purge line but even that doesn't extrude. I'm using simplify 3D with standard settings (apart from the custom start and stop gcode).

Adjusting the feedrate on the touchscreen mid print doesn't seem to have an effect.

Does anybody have an idea what this could be? Could the extruder motor not be getting enough current?

I copied the start and stop gcode from somewhere on this page, but here they are nevertheless:

;Start Gcode
G90
M140 S{material_bed_temperature_layer_0}     ;Heat bed up to first layer temperature
M190 S{material_bed_temperature_layer_0}     ;Wait for Heat Bed temperature
M104 S{material_print_temperature_layer_0}    ;Set nozzle temperature to first layer temperature
M109 S{material_print_temperature_layer_0}    ;Wait for Extruder temperature
M107 ;start with the fan off
G28

;Nozzle purge move

G1 X-70 Y-70 Z0.3 ;Move to bottom left
G1 X70 E20 F1500 ; Extrude across bed
G1 Y-69.6 ; Move up 0.4
G1 X-40 E20 F1500 ; part way back extruding
G1 X-70 F2000 ; Drain nozzle
G1 Z5 ; lift head
-------------------------------------
;end gcode

G27 ; Park nozzle
G1 Z140 F{travel_speed*60} ; lower bed to present
M140 S0 ; turn off heatbed
M104 S0 ; turn off temperature
M107 ; turn off fan
M84 X Y E ; disable motors

M73 P0 ; reset print progress
M117 Print Finished;
M300 S440 P200 ; play a tune
M300 S660 P250
M300 S880 P300 
; End G-Code

Thanks in advance

**Comment by moonglow** [comment, marked as answer]
Hi @vncntv  !

Please provide your G-code file what you try to run.

**Reply by vncntv** [reply]
Hi, github doesn't support gcode files so i saved it as a txt, I hope this is fine.
Thanks
Edit: a friend of mine told me it is better to use pastebin, so here is the link:
https://pastebin.com/RBsvHLGY

**Reply by moonglow** [reply]
Original FlashForge firmware is based on **Sailfish** firmware project. This is **Marlin** firmware and they have a little difference with Sailfish. You can't just use your old slicer settings with Marlin. You need to set Marlin capable G-Code format. As example your G-Code code use some commands what will not supported by Marlin. Another thing you also use wrong Start G-code , looks like you copied it from Cura. So you need use correct ( or use another ) slicer settings and all be OK. I never used simplify3d for my printer, because it not free software, so i think maybe someone more familiar with such slicer will can helps more, but main idea you need to config it to use with Marlin :)

**Reply by vncntv** [reply]
Hi @moonglow your supplied calibration cube didn't work either, but I do think it has to do with the fact that gcode is for another printer (because the stepperss made alot more noise than usual and didn't move ssmoothly).

I have changed the firmware from flashforge to marlin in simplify 3D but my files are still not extruding properly, i was also battling a jam so i'm going to try to use the other extruder and keep you posted.*

**Reply by moonglow** [reply]
Maybe another Dreamer users can helps you there. Really interesting for whats wrong in your case.

**Reply by vncntv** [reply]
@moonglow I've done some testing and found out that the hotend is not extruding because the motor pusshes the filament out the wrong way (it unloads basically), it does the same during purging aswell. So the motor basically spins the wrong way around.
Do you happen to know a solution to this? I was printing with the right extruder this time btw, which is extruder 1

**Reply by moonglow** [reply]
Ah... looks like FF changed logic for new and old Dreamer models.... OLD - opposite direction, NEW - same direction.... OK i need to add new builds ( because i think regular users will not use their own builds ), you can use this version for now before i will prepare new builds => #46  ( hotends swapped but all other will be OK ). Thanx for report !!!

**Reply by vncntv** [reply]
@moonglow thanks for the quick answer, I've reflashed the dreamer (with your provided swap firmware) and am using the extruding i'm certain works properly, now it seems to be printing!!!! Note for other people: if you use the swapped firmware your slicer might adress the wrong extruder, when i clicked on use left extruder it used the right, and vice versa. This can be solved by changing the identifier in your slicer (T0 and T1 have to be swapped for both left and right), make sure to also do the same for the temp sensor (in my case it didn't extrude because i was cold-extruding protecting itself). I found this out using the pronterface log. (I used the preconfigured dreamer profile by simplify 3D, and changed the firmware from sailfish to marlin).

I'm going to test out some prints and might post my simplify 3D config file if i am happy with the result, in case anybody else is struggling with the dreamer.

Many thanks to @moonglow !
